### PR TITLE
Document struct field names in stream.go

### PR DESCRIPTION
**Pull request by jnmoyne**
- Add comments to document main structure field names a bit better and improve the code accessibility.
- Rename setSourceConsumer to setupSourceConsumer (better and matches setupMirrorConsumer).

Signed-off-by: Jean-Noël Moyne <jnmoyne@gmail.com>

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -279,51 +284,58 @@ type stream struct {
...
-	compressOK bool
-	inMonitor  bool
+	sa         *streamAssignment // What the meta controller uses to assign streams to peers.
+	node       RaftNode          // Our RAFT node for the stream's group.
+	catchup    atomic.Bool       // Used to signal we are in catchup mode.
+	catchups   map[string]uint64 // The number of messages that need to be caught per peer.
+	syncSub    *subscription     // Internal subscription for sync messages (on "$JSC.SYNC").
+	infoSub    *subscription     // Internal subscription for stream info requests.
+	clMu       sync.Mutex        // The mutex for clseq and clfs.
+	clseq      uint64            // The commit log current last sequence number.
```
Comment (derekcollison) [inline server/stream.go:294]: Not the commit log, this is the run ahead of that..

**Review by derekcollison — CHANGES_REQUESTED** [review]
Generally looks good, one minor correction needed.

**Review by derekcollison — APPROVED** [review]
LGTM

**Comment by derekcollison** [comment]
Hey @jnmoyne can you rebase on top of main and resolve one conflict with your sourcing change?

**Comment by derekcollison** [comment]
Once resolved I will pull in.

**Comment by jnmoyne** [comment]
Rebased

**Pull request merged**